internal/core/http.go

```diff
@@ -129,6 +131,10 @@ func (r HTTPRoute) GetType() ResolvedRouteType {
 	return ResolvedHTTPRouteType
 }
 
+func (r HTTPRoute) Marshal() ([]byte, error) {
```

Comment:
You'll see several marshal/unmarshal additions in this PR that weren't needed up until the actual store replacement which I have completed in a separate branch